File: CinemaEpul/src/app/liste-acteurs/acteur.ts
export interface Realisateur {
    norea: number;
    nomRea: string;
    prenRea: string;
}

export interface Categorie {
    codecat: string;
    libelleCat: string;
    image: string;
}

export interface Film {
    nofilm: number;
    titre: string;
    duree: number;
    dateSortie: Date;
    budget: number;
    montantRecette: number;
    realisateur: Realisateur;
    categorie: Categorie;
    noFilm: number;
}

export interface Personnage {
    nomPers: string;
    film: Film;
}

export interface Acteur {
    noact: number;
    nomAct: string;
    prenAct: string;
    dateNaiss: Date;
    dateDeces?: any;
    personnages: Personnage[];
    noAct: number;
}

File: Mobile/lib/Model/acteur.dart
class Acteur {
  int? noact;
  String? nomAct;
  String? prenAct;
  String? dateNaiss;
  String? dateDeces;
  int? noAct;

  Acteur(
      {this.noact,
        this.nomAct,
        this.prenAct,
        this.dateNaiss,
        this.dateDeces,
        this.noAct});

  Acteur.fromJson(Map<String, dynamic> json) {
    noact = json['noact'];
    nomAct = json['nomAct'];
    prenAct = json['prenAct'];
    dateNaiss = json['dateNaiss'];
    dateDeces = json['dateDeces'];
    noAct = json['noAct'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['noact'] = this.noact;
    data['nomAct'] = this.nomAct;
    data['prenAct'] = this.prenAct;
    data['dateNaiss'] = this.dateNaiss;
    data['dateDeces'] = this.dateDeces;
    data['noAct'] = this.noAct;
    return data;
  }
}